apps/client/src/sttr_client/models/checker.js

```diff
@@ -104,6 +104,42 @@ class Checker {
     return this.stack[this.stack.length - 1];
   }
 
+  /**
+   * find all permits that need contact,
+   * for evert permit see if the decisiveQuestion for contact
+   * equals the current (recently answered) question
+   *
+   * @param {Question} currentQuestion - the question to check for exit
+   *
+   * @returns {boolean} - if checker has a 'contact' outcome
+   * consumer can exit if wanted
+   */
+  needContactExit(currentQuestion) {
+    return !!this.permits.find((permit) => {
+      const conclusion = permit.getDecisionById("dummy");
+      const conclusionMatchingRules = conclusion.getMatchingRules();
+      const matchingContactRule = conclusionMatchingRules.find(
+        (rule) => rule.outputValue === '"NeemContactOpMet"'
```

Comment:
maybe the stringish `'"NeemContactOpMet"'` should be a variable in the function making it useful for other systems as well?